HerPortal.ManagementShell/Program.cs

```diff
@@ -0,0 +1,228 @@
+                            outputProvider.Output("User not found");
+                            break;
+                        default:
+                        {
+                            {
+                                dbContext.Users.Remove(user);
+                                dbContext.SaveChanges();
+                                var deletionConfirmation = outputProvider.Confirm(
+                                    "Attention! Deletion from a database. Are you sure you want to commit this transaction? (y/n)");
+
+                                if (deletionConfirmation)
+                                {
+                                    dbContextTransaction.Commit();
+                                }
+                                else
+                                {
+                                    dbContextTransaction.Rollback();
+                                    outputProvider.Output("Rollback complete");
+                                }
+                            }
+
+                            break;
+                        }
+                    }
+                }
+                catch (Exception e)
+                {
+                    outputProvider.Output($"Rollback following error in transaction: {e.InnerException?.Message}");
+                    dbContextTransaction.Rollback();
+                }
+            }
+
+            public void CreateUserOrLogError(string userEmailAddress, string[]? custodianCodes)
+            {
+                using var dbContextTransaction = dbContext.Database.BeginTransaction();
+                try
+                {
+                    var newLaUser = new User
+                    {
+                        EmailAddress = userEmailAddress,
+                        HasLoggedIn = false,
+                        LocalAuthorities = custodianCodes!
+                            .Select(code => dbContext.LocalAuthorities
+                                .Single(la => la.CustodianCode == code))
+                            .ToList()
+                    };
+                    dbContext.Add(newLaUser);
+                    dbContext.SaveChanges();
+                    outputProvider.Output("Operation successful");
+                    dbContextTransaction.Commit();
+                }
+                catch (Exception e)
+                {
+                    outputProvider.Output($"Rollback following error in transaction: {e.InnerException?.Message}");
+                    dbContextTransaction.Rollback();
+                }
+            }
+
+            public void RemoveLasFromUser(string[]? custodianCodes, User? user)
+            {
+                var lasToRemove = user?.LocalAuthorities.Where(la => custodianCodes!.Contains(la.CustodianCode)).ToList();
+                using var dbContextTransaction = dbContext.Database.BeginTransaction();
+                try
+                {
+                    if (lasToRemove != null)
```

Comment:
Do we want some kind of warning if the number of LAs to remove is less than the number of custodian codes submitted?

Reply:
Output a warning urging the user to check the custodian codes